src/module.c

```diff
@@ -998,8 +1000,11 @@ void moduleCreateContext(ValkeyModuleCtx *out_ctx, ValkeyModule *module, int ctx
  */
 void moduleScriptingEngineInitContext(ValkeyModuleCtx *out_ctx,
                                       ValkeyModule *module,
-                                      client *client) {
-    moduleCreateContext(out_ctx, module, VALKEYMODULE_CTX_SCRIPT_EXECUTION);
+                                      client *client,
+                                      int add_thread_safe_flag) {
```

Comment:
Rather than a very specific boolean, how about `int additional_flags`?  And then pass `VALKEYMODULE_CTX_THREAD_SAFE` in?  However I don't see where that is defined.